lib/logstash/inputs/irc.rb

```diff
@@ -148,7 +148,7 @@ def handle_response (msg, output_queue)
           event.set("channel", msg.channel.to_s)
           event.set("nick", msg.user.nick)
           event.set("server", "#{@host}:#{@port}")
-          event.set("host", msg.user.host)
+          event.set("host", msg.user.data[:host]) # use unsynced host because Cinch hangs if unavailable
```

Comment:
What do you mean by `unsynced`?  Is `msg.user.host` guarded by a mutex, while `.data` is not?

Reply:
It's not as straight-forward as a mutex in implementation, but yes; `data` is a documented public method that gives access to the underlying data hash without waiting for synchronisation.

The `Cinch::User` object is essentially created by parsing the components of each IRC message, and the `host` attribute is an optional part of the message format; when it's not present, the field is not marked as synchronised, so when we send `Cinch::User#host`, it waits for the value to get populated -- but doesn't actually kick off anything _to_ populate it -- and then eventually times out and raises an exception.

If the `host` attribute is present when the message is parsed, it will be populated in the `data` Hash.

More context below:

> I took a brief dive into [cinch](https://github.com/cinchrb/cinch) and found
> its model to be rather tangled; the `Cinch::User`, `Cinch::UserList`,
> `Cinch::Bot` (which _is_ a `Cinch::User`?), and `Cinch::IRC` apparently
> attempt to work together to keep user metadata in sync, but to do so they
> juggle a variety of explicitly-given and externally-fetched data and it's not
> clear whose responsibility it is to mark portions as synced or really what
> "synced" even means.
> 
> For whatever reason, a message is being processed where the user's `host`
> attribute isn't part of the message, so it never gets marked as synced and
> times out waiting for an update that will never happen. [...]
>
> @yaauie in [Issue Comment](https://github.com/logstash-plugins/logstash-input-irc/pull/10#issuecomment-346461785)

Reply:
@yaauie that makes sense can we add just a bit more color to the comment along the lines of this note?